drag_and_drop_v2/drag_and_drop_v2.py

```diff
@@ -968,7 +1069,8 @@ def _get_user_state(self):
         if self.mode == Constants.STANDARD_MODE:
             is_finished = self._is_answer_correct()
         else:
-            is_finished = not self.attempts_remain
+            is_finished = self.is_finished
```

Comment:
Is this change acceptable?

Reply:
What if the results should be hidden (e.g. during an exam)? Wouldn't this leak the answer correctness to users?

Reply:
@Agrendalath

It will not leak the answer correctness to users as this code only applies in the assessment mode.


```is_finished ``` return true if a) the answer is correct or b) the student has used all allowed attempts or c)  the submission is past its deadline.

if this still leaks the answer correctness to users please let me know so I can change

Thanks @Agrendalath

Reply:
@abodacs, the assessment mode is what is used during exams. It could be hard to use the standard mode there.

What would happen in the following scenario?
1. Users are taking an exam. Users should not be able to see if the answer is correct until the exam is finished (is past the due date).
2. Users have multiple attempts available, so even after they submit an answer, they can still change it as long as the exam is active.
3. A user submits their answer and it's incorrect. Therefore, `is_finished` is `False`, and they can still change their answer.
4. A user submits their answer and it's correct. Therefore, `is_finished` is `True`, and they cannot change their answer anymore. They don't know the answer, but they can see it's correct because the XBlock has the `finished` state.
5. They could potentially even use the browser's Network tab to check the `finished` param in the response.

I didn't test the described scenario, but this is what I'm understanding from a quick look at the implementation. Please correct me if I'm wrong.

Reply:
@Agrendalath 

Aha, I think I got confused with the variable ```is_finished``` and the new added ```self.is_finished``` I thought there should be equal

My current understanding

- ```is_finished``` means should the user allowed to submit an answer
- ```self.is_finished``` means should the user allowed to show the answer button
- submits an answer is equal to do attempt

What would happen in the following scenario?
 _my answer when ```is_finished = self.is_finished```_

1. Users are taking an exam. Users should not be able to see if the answer is correct until the exam is finished (is past the due date).

 `is_finished` is `True` as `self.has_submission_deadline_passed` is also `True` which `self.is_finished` depends on.

2. Users have multiple attempts available, so even after they submit an answer, they can still change it as long as the exam is active.
My understanding for point 2 is that  `is_finished` is `False` as long as there are still attempts remaining and the exam is active
but this confuses me because _**they can still change it as long as the exam is active.**_  doesn't match with 3 and 4 or points 3 and 4 should be 2.a and 2.b
 
3. A user submits their answer and it's incorrect. Therefore, `is_finished` is `False`, and they can still change their answer.

`is_finished` is `False`as exam is still active and attempts remaining are available `not self.attempts_remain`

4. A user submits their answer and it's correct. Therefore, `is_finished` is `True`, and they cannot change their answer anymore. They don't know the answer, but they can see it's correct because the XBlock has the finished state.

`is_finished` is `True` as the exam is still active and the answer is correct `self.is_correct`

Reply:
solutions to that

- revert `is_finished` back to equal to  `not self.attempts_remain` and open an issue if it is wrong.
```python
   is_finished = not self.attempts_remain
```
or

- edit `is_finished` code like 
```python
   is_finished = not self.attempts_remain or self.is_correct or self.has_submission_deadline_passed
```
which is the same as `self.is_finished` but `self.is_finished ` is used for another purpose (ie, Show answer button) 

but this solution conflicts with point 2 above (in my understanding)

Of course, any other suggestions are welcomed